libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
In the case I'm talking about, there's no extra worker that receives a `nil`.  No requests have changed priority -- it's just that prefetch workers happened to have started working on on-demand requests before an on-demand worker did, leading to a situation where the on-demand worker services the prefetch requests because that's all that's left when it checks the queue.